### PR TITLE
Feature/ add verify method

### PR DESCRIPTION
**Pull request by MohammedEssehemy**
Add support for verifying pdf files that contain a ByteRange placeholder.

### CODE REVIEW

**Comment by vbuch** [comment]
Wow, @MohammedEssehemy , really a great contribution. I'll have a look at the code ASAP and probably suggest edits. Meanwhile, could you try and reach 100% of test coverage so that @coveralls doesn't complain :-)
Also, without having seen any line of code so far, just asking you to stick to the [purpose of the repo](https://github.com/vbuch/node-signpdf#purpose). You could've already written it correctly, so it is as readable as possible, but if you haven't, please, do.
As I said, I will review it as soon as possible.

File: src/helpers.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    }
+    return output;
+};
+
+export const pad2 = (num) => {
+    const s = `0${num}`;
+    return s.substr(s.length - 2);
+};
+
+export const stringToHex = (s) => {
```
[the remark this answers is not in the document]
Comment (MohammedEssehemy) [reply src/helpers.js]: the two function serve different purposes one converts from hex to binary and the other is its reverse, 
IMHO we could replace them both and use node native API as follows:
`Buffer.from(binaryString, 'binary').toString('hex');`
`Buffer.from(hexString, 'hex').toString('binary');`

[the remark this answers is not in the document]
Comment (vbuch) [reply src/helpers.js]: Yeah. I know they are reciprocal functions. I meant the names are not consistent. They should be either hexToString() and stringToHex() or hexStr() and strHex(). What you suggest is totally fine as well. I like it. It aligns well with the rest of the code, so, yes, do it, please!

File: src/helpers.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+/**
+ * Adds the objects that are needed for Adobe.PPKLite to read the signature.
+ * Also includes a placeholder for the actual signature.
+ * Returns an Object with all the added PDFReferences.
+ * @param {PDFDocument} pdf
+ * @param {string} reason
+ * @returns {object}
+ */
+export const addSignaturePlaceholder = ({pdf, reason, signatureLength = 8192}) => {
```
[the remark this answers is not in the document]
Comment (MohammedEssehemy) [reply src/helpers.js]: I thought that will be useful utility so I moved it to the helpers module and in the future we can support more pdf generation libs.

**Comment by vbuch** [comment]
I will wait for 1 more review before this gets merged just to make sure the rest f the guys have seen it. It is pretty cool at the moment. You could probably, if you have the time, add some comments on how verification is done. Also maybe some notes in the README? If you don't, I will do that once it is in develop before the new version is published.
Thank you again for the great contribution!

**Comment by MohammedEssehemy** [comment]
Thanks for your review.
I've added some notes to README

**Pull request merged**